plugins/search-indexer-backend/src/types.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Base properties that all indexed documents must include, as well as some
+ * common properties that documents are encouraged to use where appropriate.
+ */
+export interface IndexableDocument {
+  /**
+   * The primary name of the document (e.g. name, title, identifier, etc).
+   */
+  title: string;
+
+  /**
+   * Free-form text of the document (e.g. description, content, etc).
+   */
+  text: string;
```

Comment:
This makes me think that we may want to have separate input and output types, as in one type for the structure that gets indexed, and then a different type that would be returned in a search result. Typically the data you want to index likely contains a lot more than what you want and need to return in the result.

Reply:
What do you think of adding that extra/different detail as sibling properties on the `SearchResult` type [here](https://github.com/backstage/backstage/pull/4515/files#diff-5615bc9ae301d7c9018c1d62daf061b1f5344d92b042ef6b5d4e06b876c7d705R27)?  So you'd get the document, but you could also get (for example) highlight text generated by the search engine.

Reply:
My main concern here was actually passing down the entire indexed text content to the frontend. The amount of text we want to index might be quite large, and it might be a bit much compared to what we want to send to the frontend.

Thinking it might make sense conceptually too though, with there probably being a common subset for both indexed and result types, but both of them also containing their own extra fields

Reply:
Think this is something we can iterate on?  In the short-ish term, if we want to reduce the text returned on the `text` attribute, we might introduce a translation layer where we truncate or replace with highlight text, but it's still just a string, either way.

I can't think of any reason off the top of my head where we'd explicitly want something indexed, but explicitly not returned to the front-end.

Reply:
Yeah makes sense to not put anything in place before we have a clear need